pkg/library/flatten/flatten.go

```diff
@@ -315,3 +322,20 @@ func canImportDWT(dwNamespace string, dwt *dw.DevWorkspaceTemplate) bool {
 	}
 	return false
 }
+
+func FormatVariablesWarning(warn *variables.VariableWarning) string {
+	var msg []string
+	for componentName, warnings := range warn.Components {
+		msg = append(msg, fmt.Sprintf("invalid variables in component %s: '%s'", componentName, strings.Join(warnings, "', '")))
+	}
+	for commandName, warnings := range warn.Commands {
+		msg = append(msg, fmt.Sprintf("invalid variables in component %s: '%s'", commandName, strings.Join(warnings, "', '")))
+	}
+	for projectName, warnings := range warn.Projects {
+		msg = append(msg, fmt.Sprintf("invalid variables in project %s: '%s'", projectName, strings.Join(warnings, "', '")))
+	}
+	for starterProjectName, warnings := range warn.StarterProjects {
+		msg = append(msg, fmt.Sprintf("invalid variables in starter project %s: '%s'", starterProjectName, strings.Join(warnings, "', '")))
+	}
+	return fmt.Sprintf("Error processing variable replacements: %s", strings.Join(msg, "; "))
```

Comment:
when I tested I noticed that probably there are even more quotas than we would like to have
```yaml
  conditions:
  - lastTransitionTime: "2021-07-28T12:57:33Z"
    message: 'Error processing variable replacements: invalid variables in component
      vsix-installer: ''test-variable''; invalid variables in component tools: ''test-variable'''
```
Ah, I see. It's ' is used for multilines, and `''` is escaped `'`

Reply:
The joys of YAML :D 

For a while I wanted to use `%q` instead of `%s` (so that it would `print "var"` instead of `print var`), but that ends up requiring escapes all the time.